pkg/workloadbasedlearning/handle.go

```diff
@@ -32,6 +32,61 @@ func NewWorkloadBasedLearningHandle() *Handle {
 }
 
 // HandleReadTableCost Start a new round of analysis of all historical read queries.
-func (Handle *Handle) HandleReadTableCost() {
+// According to abstracted table cost metrics, calculate the percentage of read scan time and memory usage for each table.
+// The result will be saved to the table "mysql.workload_values".
+// Dataflow
+//  1. Abstract middle table cost metrics(scan time, memory usage, read frequency)
+//     from every record in statement_summary/statement_stats
+//
+// 2,3. Group by tablename, get the total scan time, total memory usage, and every table scan time, memory usage,
+//
+//	read frequency
+//
+// 4. Calculate table cost for each table, table cost = table scan time / total scan time + table mem usage / total mem usage
+// 5. Save all table cost metrics[per table](scan time, table cost, etc) to table "mysql.workload_values"
+func (handle *Handle) HandleReadTableCost() {
+	// step1: abstract middle table cost metrics from every record in statement_summary
+	middleMetrics := handle.analyzeBasedOnStatementSummary()
+	if middleMetrics == nil || len(middleMetrics) == 0 {
```

Comment:
pkg/workloadbasedlearning/handle.go:50:5: should omit nil check; len() for []*github.com/pingcap/tidb/pkg/workloadbasedlearning.ReadTableCostMetrics is defined as zero (S1009)

pkg/workloadbasedlearning/handle.go:47:45: empty-lines: extra empty line at the end of a block (all_revive)